dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/AuthorizationRestRepository.java

```diff
@@ -0,0 +1,268 @@
+     * @param featureName
+     *            limit the authorization check to only the feature identified via its name
+     * @param pageable
+     *            the pagination options
+     * @return the list of matching authorization available for the requested user and object, filtered by feature if
+     *         provided
+     * @throws AuthorizeException
+     * @throws SQLException
+     */
+    @PreAuthorize("#epersonUuid==null || hasPermission(#epersonUuid, 'EPERSON', 'READ')")
+    @SearchRestMethod(name = "object")
+    public Page<AuthorizationRest> findByObject(@Parameter(value = "uri", required = true) String uri,
+            @Parameter(value = "eperson") UUID epersonUuid,
+            Pageable pageable) throws AuthorizeException, SQLException {
+        Context context = obtainContext();
+        BaseObjectRest obj = utils.getBaseObjectRestFromUri(context, uri);
+        if (obj == null) {
+            return null;
+        }
+
+        EPerson currUser = context.getCurrentUser();
+        // get the user specified in the requested parameters
+        EPerson user = getUserFromRequestParameter(context, epersonUuid);
+        // Temporarily change the Context's current user in order to retrieve
+        // authorizations based on that user
+        context.setCurrentUser(user);
+        List<AuthorizationFeature> features = authorizationFeatureService.findByResourceType(obj.getUniqueType());
+        List<Authorization> authorizations = new ArrayList<Authorization>();
+        for (AuthorizationFeature f : features) {
+            if (authorizationFeatureService.isAuthorized(context, f, obj)) {
+                authorizations.add(new Authorization(user, f, obj));
+            }
+        }
+        // restore the real current user
+        context.setCurrentUser(currUser);
+        return converter.toRestPage(utils.getPage(authorizations, pageable), utils.obtainProjection());
+    }
+
+    /**
+     * It returns the authorization related to the requested feature if granted to the specified eperson or to the
+     * anonymous user. Only administrators and the user identified by the epersonUuid parameter can access this method
+     *
+     * 
+     * @param context
+     *            the DSpace Context
+     * @param uri
+     *            the uri of the object to check the authorization against
+     * @param epersonUuid
+     *            the eperson uuid to use in the authorization evaluation
+     * @param featureName
+     *            limit the authorization check to only the feature identified via its name
+     * @param pageable
+     *            the pagination options
+     * @return the list of matching authorization available for the requested user and object, filtered by feature if
+     *         provided
+     * @throws AuthorizeException
+     * @throws SQLException
+     */
+    @PreAuthorize("#epersonUuid==null || hasPermission(#epersonUuid, 'EPERSON', 'READ')")
+    @SearchRestMethod(name = "objectAndFeature")
```

Comment:
why is this a separate endpoint? According to the contract this should be part of the /object endpoint, and the feature is an optional parameter

Reply:
in the contract update PR I have included this change. It makes much easier and clean the implementation and the testing

Reply:
The impact for Angular is most likely limited, so this should work as well